server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/QualifiedReferenceUpdateVariableUsage.java

```diff
@@ -41,38 +42,36 @@ public class QualifiedReferenceUpdateVariableUsage implements Processor<Qualifie
   private static final String NOT_DEFINED_ERROR = "semantics.notDefined";
   private static final String AMBIGUOUS_REFERENCE_ERROR = "semantics.ambiguous";
 
-  private final SymbolAccumulatorService symbolAccumulatorService;
+  private final SymbolAccumulator symbolAccumulator;
 
-  public QualifiedReferenceUpdateVariableUsage(SymbolAccumulatorService symbolAccumulatorService) {
-    this.symbolAccumulatorService = symbolAccumulatorService;
+  public QualifiedReferenceUpdateVariableUsage(SymbolAccumulator symbolAccumulator) {
+    this.symbolAccumulator = symbolAccumulator;
   }
 
   @Override
   public void accept(QualifiedReferenceNode node, ProcessingContext ctx) {
-    List<VariableUsageNode> variableUsageNodes =
-            node.getChildren().stream()
-                    .filter(Node.hasType(NodeType.VARIABLE_USAGE))
-                    .map(VariableUsageNode.class::cast)
-                    .collect(Collectors.toList());
+    List<VariableUsageNode> variableUsageChain = new ArrayList<>();
```

Comment:
What makes it a chain?

Reply:
Path?

Reply:
I don't quite understand how the subset of the children gets turned into a chain/path.

Reply:
From what I understand it's a list of parents variables nodes so we can identify the variable in COBOL code.